packages/nocodb/src/lib/version-upgrader/ncFilterUpgrader_0104004.ts

```diff
@@ -0,0 +1,380 @@
+import { NcUpgraderCtx } from './NcUpgrader';
+import { MetaTable } from '../utils/globals';
+import NcMetaIO from '../meta/NcMetaIO';
+import Column from '../models/Column';
+import Filter from '../models/Filter';
+import Project from '../models/Project';
+import { UITypes, SelectOptionsType } from 'nocodb-sdk';
+
+// as of 0.104.3, almost all filter operators are available to all column types
+// while some of them aren't supposed to be shown
+// this upgrader is to remove those unsupported filters / migrate to the correct filter
+
+// Change Summary:
+// - Text-based columns:
+//     - remove `>`, `<`, `>=`, `<=`
+//     - Numeric-based / SingleSelect columns:
+//     - remove `like`
+//     - migrate `null`, and `empty` to `blank`
+// - Checkbox columns:
+//     - remove `equal`
+//     - migrate `empty` and `null` to `notchecked`
+// - MultiSelect columns:
+//     - remove `like`
+//     - migrate `equal`, `null`, `empty`
+//     - Attachment columns:
+//     - remove `>`, `<`, `>=`, `<=`, `equal`
+//     - migrate `empty`, `null` to `blank`
+// - LTAR columns:
+//     - remove `>`, `<`, `>=`, `<=`
+//     - migrate `empty`, `null` to `blank`
+// - Lookup columns:
+//     - migrate `empty`, `null` to `blank`
+// - Duration columns:
+//     - remove  `like`
+//     - migrate `empty`, `null` to `blank`
+
+const removeEqualFilters = async (filter, actions: any[], ncMeta) => {
+  // remove `is equal`, `is not equal`
+  if (['eq', 'neq'].includes(filter.comparison_op)) {
+    actions.push(await Filter.delete(filter.id, ncMeta));
+  }
+  return actions;
+};
+
+const removeArithmeticFilters = async (filter, actions: any[], ncMeta) => {
+  // remove `>`, `<`, `>=`, `<=`
+  if (['gt', 'lt', 'gte', 'lte'].includes(filter.comparison_op)) {
+    actions.push(await Filter.delete(filter.id, ncMeta));
+  }
+  return actions;
+};
+
+const removeLikeFilters = async (filter, actions: any[], ncMeta) => {
+  // remove `is like`, `is not like`
+  if (['like', 'nlike'].includes(filter.comparison_op)) {
+    actions.push(await Filter.delete(filter.id, ncMeta));
+  }
+  return actions;
+};
+
+const migrateNullAndEmptyToBlankFilters = async (
+  filter,
+  actions: any[],
+  ncMeta
+) => {
+  if (['empty', 'null'].includes(filter.comparison_op)) {
+    // migrate to blank
+    actions.push(
+      await Filter.update(
+        filter.id,
+        {
+          ...filter,
```

Comment:
you can remove the `...filter,` since we are just updating `comparison_op`

Reply:
Oh got it in function level we are doing 

```
   const updateObj = {
      fk_column_id: filter.fk_column_id,
      comparison_op: filter.comparison_op,
      value: filter.value,
      fk_parent_id: filter.fk_parent_id,

      is_group: filter.is_group,
      logical_op: filter.logical_op,
    };
```


We can replace it with something like

```
 const updateObj = extractProps(base, [
      'alias',
      'config',
      'type',
      'is_meta',
      'created_at',
      'updated_at',
      'inflection_column',
      'inflection_table',
      'order',
      'enabled',
    ]);
```